src/frontend/admin/src/components/presentational/discount/DiscountSelect.tsx

```diff
@@ -0,0 +1,262 @@
+    description: "The button to create a new discount",
+  },
+  createDiscountDialogTitle: {
+    id: "components.presentational.discount.DiscountSelect.createDiscountDialogTitle",
+    defaultMessage: "Create new discount",
+    description: "The title of the create discount dialog",
+  },
+  amountLabel: {
+    id: "components.presentational.discount.DiscountSelect.amountLabel",
+    defaultMessage: "Amount (€)",
+    description: "The label for the amount field",
+  },
+  rateLabel: {
+    id: "components.presentational.discount.DiscountSelect.rateLabel",
+    defaultMessage: "Rate (%)",
+    description: "The label for the rate field",
+  },
+  cancelButton: {
+    id: "components.presentational.discount.DiscountSelect.cancelButton",
+    defaultMessage: "Cancel",
+    description: "The cancel button",
+  },
+  createButton: {
+    id: "components.presentational.discount.DiscountSelect.createButton",
+    defaultMessage: "Create",
+    description: "The create button",
+  },
+  discountHelperText: {
+    id: "components.presentational.discount.DiscountSelect.discountHelperText",
+    defaultMessage: "Select a discount or create a new one",
+    description: "The helper text for the discount select",
+  },
+  discountRateHelperText: {
+    id: "components.presentational.discount.DiscountSelect.discountRateHelperText",
+    defaultMessage: "Enter a rate between 0 and 100",
+    description: "The helper text for the rate field",
+  },
+  discountAmountHelperText: {
+    id: "components.presentational.discount.DiscountSelect.discountAmountHelperText",
+    defaultMessage: "Enter a fixed amount in euros",
+    description: "The helper text for the amount field",
+  },
+  discountExclusiveHelperText: {
+    id: "components.presentational.discount.DiscountSelect.discountExclusiveHelperText",
+    defaultMessage: "You can only set either a rate or an amount, not both",
+    description:
+      "The helper text explaining that rate and amount are exclusive",
+  },
+});
+
+type Props = {
+  value?: string | null;
+  onChange: (discountId: string | null) => void;
+  error?: boolean;
+  helperText?: string;
+};
+
+export function DiscountSelect({ value, onChange, error, helperText }: Props) {
+  const intl = useIntl();
+  const [discounts, setDiscounts] = useState<Discount[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const [newDiscount, setNewDiscount] = useState<DTODiscount>({
+    amount: null,
+    rate: null,
+  });
+  const [formError, setFormError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchDiscounts = async () => {
+      try {
+        const response = await DiscountRepository.getAll();
+        setDiscounts(response.results);
+      } catch (fetchError) {
+        setFormError(`Failed to fetch discounts: ${fetchError}`);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchDiscounts();
+  }, []);
```

Comment:
To interact with API we are currently using react-query so we are creating a hook (e.g: `useDiscount`) base on `useQuery` and `useMutation`. You cant take a look at 'useResource' hook.